src/Centreon/Domain/PlatformTopology/Interfaces/PlatformTopologyReadRepositoryInterface.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ *
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Centreon\Domain\PlatformTopology\Interfaces;
+
+use Centreon\Domain\PlatformTopology\Platform;
+
+interface PlatformTopologyReadRepositoryInterface
+{
+    /**
+     * Search for already registered servers using same name or address
+     *
+     * @param string $address
+     * @param string $name
+     * @return bool returns true if a server is already registered using same address or name
+     */
+    public function isPlatformAlreadyRegisteredInTopology(string $address, string $name): bool;
```

Comment:
I think this method could be into the PlatformTopologyService:

We already have a method findPlatformByAddress, so we could considerate to add a method findPlatformByName, and the service will call both of those methods, if any result are found, return true, else return false.

So the Repository could be entirely dedicate to Read Data when asked, and not delegated to check informations.

Reply:
ok